### ISSUE TITLE
Solution for all vlingo-net modules in order to build everything together

### ISSUE THREAD

**Issue by tjaskula** [vlingo-net/vlingo-net-platform#1]
The goal is to have a central .NET solution for all `vlingo-net-*` modules. Currently all modules are listed here https://github.com/vlingo-net.

# 1. Pulling sources from related repositories

For that I think we need some sort of scripts that would work both on Windows and *Nix platforms.
If I clone `vlingo-net-platform` I would like to be able later to update sources in my local repository. Through scripts or other means I would need to pull/fetch updates from remote branches of each https://github.com/vlingo-net repository. Constraints:

- [ ] it should work on Windows and *Nix
- [ ] Able to pull/fetch
- [ ] Able to push
- [ ] Should work on forks

# 2. Solutions sln and Projects csproj

So this repo would only contain .sln and various csproj in order to be able to build all modules together. We cannot use the original csproj in each repository because  they reference nuget packages. So we have to be able in step 1) to somehow deal with that problem. In this repo, all projects has to have direct dependencies to other modules and not nuget packages.

Maybe I'm overlooking something. Maybe there is a better approach to achieve it. Let's discuss it here.

**Comment by jrouaix** [comment]
Hello @tjaskula, couldn't you just move all projects into a single repo ? then building and publishing multiple dependent nuget packages from a mono repo is just brain dead simple ...

**Comment by tjaskula** [comment]
@jrouaix no, it's not the goal.

**Comment by mihaj** [comment]
@tjaskula I have a few questions to get this straight.

1. The idea is to generate a *.sln with *.csproj files where no nuget vlingo dependencies are used. csproj all reference each other.
2. When generating sln and csproj files, we need to specify if the libraries should be pulled from github repositories or local github repositories, branches and forms? This way we can configure the generator and use Git library to get the files.
3. What do you mean by pushing the code? For example, you can edit the actor library from the plaftform solution and you want to push the files back to local /remote library repository?

Current PR suggests the use of Nuke builder and git submodules. This looks promising but needs further investigation.

**Comment by jrouaix** [comment]
> 
> 
> @jrouaix no, it's not the goal.

@tjaskula, sorry to bother, but again, I think you try to tackle a problem you just shouldn't have.

A monorepo will fix all your "**goals**":
- an all in one solution
- working with forks
- having nuget dependencies when publishing nugets while having project references while working on the code.

Only if you plan to be able to "version bump" projects independently .. perhaps is this the goal you wanna achieve ? otherwise I don't see it. Could you explain the goal of all this ?

**Comment by jrouaix** [comment]
Perhaps @VaughnVernon ? you have an idea about the "why" behind all this ?
Is this to be aligned with the java practices ? having some kind of 1<=>1 bijection between the implementations ?

**Comment by VaughnVernon** [comment]
@jrouaix I don't have strong opinions on this and the simpler the better. However, we have found in the Java platform that nested/submodule repos don't work out well. This has a lot to do with out CI/CD toolchain. (Currently our CI is not the greatest, but it works. We will improve this.) For example from Travis-CI to Bintray and Bintray to release repositories Sonatype and Maven Central the multiple artifacts per git repo causes big problems.

Another concern is that once you start moving things around and into one repo you lose all history. I have tried to retain history and several have claimed that it can be done, but IMO it really cannot.
